test/TestProviderRound.spec.js

```diff
@@ -168,32 +216,32 @@ contract('ProviderRound', accounts => {
     });
 
     it('should increase the totalAmountBonded of the provider', async () => {
-      let firstProvider = await providerRound.providers.call(accounts[0]);
-      assert.equal(10, firstProvider[5]);
       await providerRound.approve(contractAddress, 20, {from: accounts[6]});
-      await providerRound.bond(accounts[1], 20, {from: accounts[6]});
-      await providerRound.approve(contractAddress, 50, {from: accounts[7]});
-      await providerRound.bond(accounts[1], 50, {from: accounts[7]});
-      firstProvider = await providerRound.providers.call(accounts[0]);
-      const secondProvider = await providerRound.providers.call(accounts[1]);
-      assert.equal(10, firstProvider[5]); // [5] is totalAmountBonded
-      assert.equal(70, secondProvider[5]); // [5] is totalAmountBonded
+      await providerRound.bond(accounts[0], 20, {from: accounts[6]});
+      const provider = await providerRound.providers.call(accounts[0]);
+      assert.equal(31, provider[5].toNumber()); // [5] is totalAmountBonded
+    });
+
+    it('should fail if the address is not a registered provider address', async () => {
+      await providerRound.approve(contractAddress, 15, {from: accounts[7]})
+      await assertFail( providerRound.bond(accounts[2], 15, {from: accounts[7]}) );
     });
 
-    it('should fail if no providerCandidate is associated with the given providerCandidateId', async () => {
-      await assertFail( providerRound.bond(accounts[2], 15, {from: accounts[5]}) );
+    it('should work if the address is not a registered provider address but the address is the sender address', async () => {
+      await providerRound.approve(contractAddress, 15, {from: accounts[3]})
+      await providerRound.bond(accounts[3], 15, {from: accounts[3]});
     });
 
     it('should fail if called twice by the same delegator', async () => {
       await providerRound.approve(contractAddress, 40, {from: accounts[8]});
-      await providerRound.bond(accounts[1], 40, {from: accounts[8]});
-      await assertFail( providerRound.bond(accounts[1], 40, {from: accounts[8]}) );
+      await providerRound.bond(accounts[1], 20, {from: accounts[8]});
+      await assertFail( providerRound.bond(accounts[1], 20, {from: accounts[8]}) );
     });
 
     it('should fail if TST balance is less than bonded amount', async () => {
       await providerRound.approve(contractAddress, 1001, {from: accounts[9]});
       await assertFail( providerRound.bond(accounts[1], 1001, {from: accounts[9]}) )
-      assert.equal(110, (await providerRound.providers(accounts[1]))[5]);
+      assert.equal(21, (await providerRound.providers(accounts[1]))[5]);
```

Comment:
This really just needs to be an assertion that `1001` is greater than `providerRound.providers(accounts[1]))[5]` instead of asserting that `providerRound.providers(accounts[1]))[5]` is equal to `21`.